packages/loader/container-loader/src/connectionManager.ts

```diff
@@ -453,6 +475,9 @@ export class ConnectionManager implements IConnectionManager {
             if (this.closed) {
                 throw new Error("Attempting to connect a closed DeltaManager");
             }
+            if (abortSignal.aborted === true) {
+                return;
+            }
             connectRepeatCount++;
 
             try {
```

Comment:
Is there any risk of concurrent `connectCore` calls conflicting on `this.client`?  If `docService.connectToDeltaStream` reads `client.mode` after awaiting at any point, the value could have been changed by the other `connectCore` (if the modes don't match).

Maybe instead of overwriting `this.client.mode` we pass in `{ ...this.client, mode: requestedMode }`, if it's safe to clone `this.client` in this way.